src/ocean/core/array/Search.d

```diff
@@ -1278,7 +1278,9 @@ unittest
 
 bool startsWith ( T ) ( in T[] arr, in T[] prefix )
 {
-    return (arr.length >= prefix.length) && (arr[0..prefix.length] == prefix[]);
+    if (arr.length < prefix.length)
+        return false;
+    return arr[0..prefix.length] == prefix[];
```

Comment:
I would like to see a better explanation for this change in the commit message, at simple sight it looks like rearranging the code. So what's the real problem and what's the intended change should be stated in the commit message.

Reply:
It is simply re-arranging code indeed. Problem is major code generation bug in compiler (can't find bugzilla number right now). Looks like it doesn't do early return from `&&` expression here when both `-O` and `-cov` are supplied.

Reply:
I strongly suggest adding a comment here too, otherwise someone could try to rearrange the code again in the future. Ideally a comment with a link to the upstream bug.

Reply:
Done